src/main/java/com/epam/aidial/core/util/EventStreamParser.java

```diff
@@ -0,0 +1,178 @@
+package com.epam.aidial.core.util;
+
+import com.epam.aidial.core.function.BaseResponseFunction;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.netty.buffer.ByteBuf;
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.buffer.impl.VertxByteBufAllocator;
+import jodd.io.CharBufferReader;
+import lombok.extern.slf4j.Slf4j;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+public class EventStreamParser {
+
+    private static final byte[] EVENT_TOKEN = "data: ".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] DONE_TOKEN = "[DONE]".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] BOM = new byte[]{(byte) 0xEF, (byte) 0xBB, (byte) 0xBF};
+
+    private static final BaseResponseFunction DEFAULT_HANDLER = new BaseResponseFunction(null, null) {
+        @Override
+        public Future<Void> apply(ObjectNode jsonNodes) {
+            return Future.succeededFuture();
+        }
+    };
+
+    private final ByteBuf buffer;
+
+    private int eventIndex;
+
+    private int chunkIndex;
+
+    private boolean firstChunk = true;
+
+    private boolean lastChunk = false;
+
+    private final BaseResponseFunction handler;
+
+    private Stages stage;
+
+    private List<Future<Void>> futures;
+
+    private enum Stages {
+        EVENT, DATA, EOL
+    }
+
+
+    public EventStreamParser(int initialSizeHint, BaseResponseFunction handler) {
+        this.handler = handler == null ? DEFAULT_HANDLER : handler;
+        buffer = VertxByteBufAllocator.DEFAULT.heapBuffer(initialSizeHint, Integer.MAX_VALUE);
```

Comment:
why not just ByteBufAllocator.allocate()

Reply:
fixed